tuning/utils/preprocessing_utils.py

```diff
@@ -0,0 +1,329 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Standard
+from typing import Any, Callable, Dict, Optional
+import json
+
+# Third Party
+from datasets import Dataset
+from transformers import AutoTokenizer, DataCollatorForSeq2Seq
+from trl import DataCollatorForCompletionOnlyLM
+import datasets
+
+# Local
+from tuning.config import configs
+
+
+def get_data_trainer_kwargs(
```

Comment:
I am just wondering if this wrapper function is hiding too much from main code. Wondering if it would make sense to move this back to main code, esp once the hack is removed . as its basically 2 steps ->
 get collator, get_formatted_dataset -> internally data formatting might happen in different ways. I am still thinking about it, but on initial thoughts I feel it might be good to know high level steps in train()